update.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+import github
+
+import argparse
+import distutils.util
+import glob
+import os
+import shutil
+import subprocess
+import sys
+import tarfile
+import urllib
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+	"--github-access-token",
+	dest = "githubAccessToken",
+	default = os.environ.get( 'GITHUB_ACCESS_TOKEN', None ),
+	help = "A suitable access token to authenticate the GitHub API."
+)
+parser.add_argument(
+	"--set-outputs",
+	dest = "setOutputs",
+	type = distutils.util.strtobool,
+	default = "0",
+	help = "Echo the outputs needed by `.github/workflows/update.yml`."
+)
+args = parser.parse_args()
+
+def __versionFromString( s ) :
+
+	# We use a list of ints for the version, as it
+	# can be compared usefully.
+	try :
+		return [ int( n ) for n in s.split( "." ) ]
+	except :
+		# We don't care about non-numeric releases
+		# like betas etc.
+		return [ 0, 0, 0, 0 ]
+
+def __versionToString( v ) :
+
+	return ".".join( str( s ) for s in v )
+
+# Find the latest release
+
+client = github.Github( args.githubAccessToken )
+repo = client.get_repo( "GafferHQ/gaffer" )
+
+latestRelease = None
+latestReleaseVersion = __versionFromString( "0.0.0.0" )
+
+for release in repo.get_releases() :
+
+	if release.prerelease :
+		continue
+
+	releaseVersion = __versionFromString( release.tag_name )
+	if releaseVersion > latestReleaseVersion :
+		latestRelease = release
+		latestReleaseVersion = releaseVersion
+
+# # If its up to date with the latest documented version, then we're done
+
+latestDocVersion = max( [ __versionFromString( v ) for v in glob.glob( "*.*.*.*" ) ] )
```

Comment:
Given our past experience with failed deploys, Is it worth us checking how long `glob.glob(...)` is, and warning/erroring if there are more than _n_ versions already present.

Reply:
Definitely, but I deliberately didn't tackle that this time. I will automate next time we hit the problem, when we can figure out what `n` should be (I presume it's not versions, but overall number of files or file size).